app/src/main/java/com/example/mykinopoisk/presentation/ui/movie/MovieDetailViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.mykinopoisk.presentation.ui.movie
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.mykinopoisk.domain.usecase.movie.GetMovieDetailUseCase
+import com.example.mykinopoisk.presentation.constants.Constants
+import com.example.mykinopoisk.presentation.model.LceState
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.stateIn
+
+class MovieDetailViewModel(
+    private val getMovieDetailUseCase: GetMovieDetailUseCase,
+    private val movieId: Long
+) : ViewModel() {
+
+    val movieDetailFlow = flow {
+        delay(2000)
+        val state = getMovieDetailUseCase.invoke(
```

Comment:
мы использовали `operator fun invoke()` как раз чтобы можно было на объектах вызывать функции как лямбды, то есть так 
```suggestion
        val state = getMovieDetailUseCase(
```